From d8d3a570fde3799c2933be25edd52be7448aeac7 Mon Sep 17 00:00:00 2001
From: Vratislav Podzimek <v.podzimek@mykolab.com>
Date: Tue, 14 May 2024 16:52:38 +0200
Subject: [PATCH 2/4] Define _XOPEN_SOURCE_EXTENDED as 1

Otherwise build on AIX 7 hits this beauty:

```
16:38:25 /usr/include/standards.h:88:28: error: operator '==' has no left operand
16:38:25  #if (_XOPEN_SOURCE_EXTENDED==1)
16:38:25                             ^~
```

(cherry picked from commit c762f970d1c390908def261001a9d2873014bbef)
---
 apps/ocsp.c            | 2 +-
 crypto/conf/conf_lib.c | 2 ++
 crypto/conf/conf_sap.c | 2 ++
 3 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/apps/ocsp.c b/apps/ocsp.c
index 97454a4b81..299323390c 100644
--- a/apps/ocsp.c
+++ b/apps/ocsp.c
@@ -11,7 +11,7 @@
 
 #ifdef OPENSSL_SYS_VMS
   /* So fd_set and friends get properly defined on OpenVMS */
-# define _XOPEN_SOURCE_EXTENDED
+# define _XOPEN_SOURCE_EXTENDED 1
 #endif
 
 #include <stdio.h>
diff --git a/crypto/conf/conf_lib.c b/crypto/conf/conf_lib.c
index 601f494302..0665644cde 100644
--- a/crypto/conf/conf_lib.c
+++ b/crypto/conf/conf_lib.c
@@ -7,6 +7,8 @@
  * https://www.openssl.org/source/license.html
  */
 
+#define _XOPEN_SOURCE_EXTENDED 1  /* To get a definition of strdup() */
+
 #include "internal/e_os.h"
 #include <stdio.h>
 #include <string.h>
diff --git a/crypto/conf/conf_sap.c b/crypto/conf/conf_sap.c
index 6b3defe0fc..d967a4cb5a 100644
--- a/crypto/conf/conf_sap.c
+++ b/crypto/conf/conf_sap.c
@@ -7,6 +7,8 @@
  * https://www.openssl.org/source/license.html
  */
 
+#define _XOPEN_SOURCE_EXTENDED 1   /* To get a definition of strdup() */
+
 #include <stdio.h>
 #include <openssl/crypto.h>
 #include "internal/cryptlib.h"
-- 
2.45.0

